presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -1847,6 +1849,211 @@ public void testDropNonExistentRole()
+
+        assertUpdate("REVOKE grant_revoke_role_3 FROM ROLE grant_revoke_role_1");
+
+        actual = queryRunner.execute(user1, "SELECT * FROM hive.information_schema.applicable_roles");
+        expected = MaterializedResult.resultBuilder(user1, createUnboundedVarcharType(), createUnboundedVarcharType(), createUnboundedVarcharType(), createUnboundedVarcharType())
+                .row("grant_revoke_user_1", "USER", "admin", "NO")
+                .row("grant_revoke_user_1", "USER", "public", "NO")
+                .row("grant_revoke_user_1", "USER", "grant_revoke_role_1", "NO")
+                .row("grant_revoke_role_1", "ROLE", "grant_revoke_role_2", "NO")
+                .build();
+        assertEqualsIgnoreOrder(actual, expected);
+
+        assertUpdate("GRANT grant_revoke_role_3 TO ROLE grant_revoke_role_1");
+
+        actual = queryRunner.execute(user1, "SELECT * FROM hive.information_schema.applicable_roles");
+        expected = MaterializedResult.resultBuilder(user1, createUnboundedVarcharType(), createUnboundedVarcharType(), createUnboundedVarcharType(), createUnboundedVarcharType())
+                .row("grant_revoke_user_1", "USER", "admin", "NO")
+                .row("grant_revoke_user_1", "USER", "public", "NO")
+                .row("grant_revoke_user_1", "USER", "grant_revoke_role_1", "NO")
+                .row("grant_revoke_role_1", "ROLE", "grant_revoke_role_2", "NO")
+                .row("grant_revoke_role_1", "ROLE", "grant_revoke_role_3", "NO")
+                .build();
+        assertEqualsIgnoreOrder(actual, expected);
+
+        assertUpdate("DROP ROLE grant_revoke_role_3");
+
+        actual = queryRunner.execute(user1, "SELECT * FROM hive.information_schema.applicable_roles");
+        expected = MaterializedResult.resultBuilder(user1, createUnboundedVarcharType(), createUnboundedVarcharType(), createUnboundedVarcharType(), createUnboundedVarcharType())
+                .row("grant_revoke_user_1", "USER", "admin", "NO")
+                .row("grant_revoke_user_1", "USER", "public", "NO")
+                .row("grant_revoke_user_1", "USER", "grant_revoke_role_1", "NO")
+                .row("grant_revoke_role_1", "ROLE", "grant_revoke_role_2", "NO")
+                .build();
+        assertEqualsIgnoreOrder(actual, expected);
+
+        assertUpdate("DROP ROLE grant_revoke_role_1");
+
+        actual = queryRunner.execute(user1, "SELECT * FROM hive.information_schema.applicable_roles");
+        expected = MaterializedResult.resultBuilder(user1, createUnboundedVarcharType(), createUnboundedVarcharType(), createUnboundedVarcharType(), createUnboundedVarcharType())
+                .row("grant_revoke_user_1", "USER", "admin", "NO")
+                .row("grant_revoke_user_1", "USER", "public", "NO")
+                .build();
+        assertEqualsIgnoreOrder(actual, expected);
+
+        assertUpdate("DROP ROLE grant_revoke_role_2");
+    }
+
+    @Test
+    public void testGrantRoleWithGrantOption()
```

Comment:
Ditto. Break this into smaller tests to better express your intention, what you exactly test and to emphasize isolation (not depending on side effects of previous tests).